### ISSUE TITLE
Design token generation tools

### ISSUE THREAD

**Issue by rav-pradhan** [ONSdigital/dp-design-system#20]
Something that was picked up on recently was how we do not have a single source of truth for our design tokens. In the repo currently, we have our variables set in Sass. However, for the colours documentation page we have had to lift the colours variables and convert them into a JS module for ease-of-use in 11ty.

I think it could be useful for us to consider storing all design tokens (variables) in a language agnostic format - YAML/JSON - and then leverage a design token generation tool that could let us translate those files into the formats we require. 

I imagine there could already be a tool out there we could use, but if not, there may be value in building one ourselves.

**Comment by armstrongb** [comment]
https://heydonworks.com/article/design-tokens-in-eleventy/ (you'll need js off as it's Heydon's site) has some info how to use design tokens in 11ty...

**Comment by rav-pradhan** [comment]
> https://piccalil.li/tutorial/what-are-design-tokens/ - plenty of reading materials and token tools listed here... going to take a bit of evaluation though to pick the right path.

Thanks, Andy's article in particular looks pertinent. Trevor mentioned Theo to me yesterday, so I'm hoping at some point in the coming weeks I / another dev will take a look at how it works. The Sketch/Figma plugins look quite handy as a quick way to produce a universal JSON for us to use, as well

**Comment by designorant** [comment]
Shouldn't this be raised over at https://github.com/ONSdigital/design-system?

I'm over at Find Insights where we loosely ported parts of the Design System into Svelte, and the only way to build upon it while staying respectful to all the base variables (without duplicating them) is to stick to Sass and import `_vars.scss` directly.

Unfortunately Sass is where the build performance takes a massive hit compared that we cannot really afford. We're settling on PostCSS for performance and ecosystem which is why I'm copying everything over manually with a long term goal to automate this process to stay synchronised upstream (not that I expect any fundamental changes often but still, Design System is a living organism after all, and redesigns and rebrands do happen).

Long story short, a JSON based list of tokens available in `@ons/design-system` npm module would be amazing.
